File: ClinicaManagement/ClinicaManagement/ClinicaDA/DAO/MedicoDAO.cs
using ClinicaModel;
using System.Collections.Generic;

namespace ClinicaDA.DAO
{
    public interface MedicoDAO
    {
        int insertar(Medico medico);
        int modificar(Medico medico);
        int eliminar(int idMedico);
        List<Medico> listar();
        Medico obtenerPorId(int idMedico);

        List<Medico> obtenerPorEspecialidad(string Especialidad);
        Usuario ExisteUsuario(string username, string contrasenha);
    }
}

File: Proyecto Java/SoftCapsuleCareController/src/pe/edu/pucp/citamedica/dao/MedicoDAO.java
    package pe.edu.pucp.citamedica.dao;
import java.util.ArrayList;
import pe.edu.pucp.citamedica.model.clinica.Medico;
import pe.edu.pucp.citamedica.model.usuario.Usuario;

/**
 *
 * @author berna
 */
public interface MedicoDAO {
    int insertar(Medico medico, Usuario usuario);
    int modificar(Medico medico);
    int eliminar(int idMedico);
    ArrayList<Medico> listarTodos();
    Medico obtenerPorId(int idMedico);
    ArrayList<Medico> listarPorEspecialidad(String especialidad);
    ArrayList<Medico> listarTodos1();
}
